pkg/catalogv2/system/system.go

```diff
@@ -42,6 +40,13 @@ var (
 			"system:masters",
 		},
 	}
+
+	// noVersionRefresh maps charts which should be updated conditionally when Rancher restarts, to version settings
```

Comment:
I agree with the idea of having this kind of chart-specific logic moved to the corresponding chart-specific controller, but, apart from making this behaviour available for other charts in the future (which admittedly may not be necessary), having it here enables another use case to be supported.
And this is where the PR description is indeed misleading: we need to prevent Fleet updates made via the Rancher UI from being undone when [system charts are refreshed](https://github.com/rancher/rancher/blob/release/v2.7/pkg/catalogv2/system/system.go#L144), ie every 6 hours in production.

Reply:
>Fleet updates made via the Rancher UI

How is this done today? I am unfamiliar with this workflow since users cannot update settings via Rancher UI if they are set via environmental variables.

Reply:
This can be done by:
1. Clicking the hamburger button on the top-left
2. Selecting a cluster in `EXPLORE CLUSTER`
3. Selecting Apps → Installed Apps in the sidebar
4. Ensuring that `All Namespaces` are selected via the upper drop-down list
5. Clicking on `fleet` in the `Installed Apps` list
6. Clicking the button in the upper right corner with a vertical line of 3 dots, then `Edit/Upgrade`
7. Selecting a fleet version to install on the following screen, then following the update process:
![Screenshot_20230817_160056](https://github.com/rancher/rancher/assets/2477066/a6eb1ad2-3e97-46ea-b421-b2b907f7c104)